File: update_title.py
import mysql.connector
import os

# Configurações do banco
db_config = {
    "host": "localhost",
    "user": "faculdade_user",
    "password": "Faculdade2024!",
    "database": "faculdade_db"
}

novo_titulo = "Empresários Educacionais: transformem propósito em rentabilidade real"

# 1. Atualizar no Banco de Dados
try:
    conn = mysql.connector.connect(**db_config)
    cursor = conn.cursor()
    
    query = "UPDATE home_settings SET value = %s WHERE section = 'about' AND field = 'title'"
    cursor.execute(query, (novo_titulo,))
    conn.commit()
    print(f"Banco de dados atualizado: {cursor.rowcount} linha(s) afetada(s).")
    
    cursor.close()
    conn.close()
except Exception as e:
    print(f"Erro ao atualizar banco: {e}")

# 2. Atualizar no Código-Fonte (Fallback)
filepath = "/var/www/faculdade-site/client/src/components/About.tsx"
try:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    old_text = "Empresários Educacionais: transformem propósito em rentabilidade real com o maior ecossistema de cursos do mundo"
    new_content = content.replace(old_text, novo_titulo)
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Código-fonte atualizado: {filepath}")
except Exception as e:
    print(f"Erro ao atualizar código: {e}")

File: update_title.mjs
import mysql from 'mysql2/promise';
import fs from 'fs';

const dbConfig = {
    host: 'localhost',
    user: 'faculdade_user',
    password: 'Faculdade2024!',
    database: 'faculdade_db'
};

const novoTitulo = "Empresários Educacionais: transformem propósito em rentabilidade real";

async function update() {
    // 1. Atualizar no Banco de Dados
    try {
        const connection = await mysql.createConnection(dbConfig);
        const [result] = await connection.execute(
            "UPDATE home_settings SET value = ? WHERE section = 'about' AND field = 'title'",
            [novoTitulo]
        );
        console.log(`Banco de dados atualizado: ${result.affectedRows} linha(s) afetada(s).`);
        await connection.end();
    } catch (error) {
        console.error(`Erro ao atualizar banco: ${error.message}`);
    }

    // 2. Atualizar no Código-Fonte (Fallback)
    const filepath = "/var/www/faculdade-site/client/src/components/About.tsx";
    try {
        let content = fs.readFileSync(filepath, 'utf8');
        const oldText = "Empresários Educacionais: transformem propósito em rentabilidade real com o maior ecossistema de cursos do mundo";
        
        if (content.includes(oldText)) {
            const newContent = content.replace(oldText, novoTitulo);
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Código-fonte atualizado: ${filepath}`);
        } else {
            console.log("Texto antigo não encontrado no código-fonte (pode já ter sido alterado via banco).");
        }
    } catch (error) {
        console.error(`Erro ao atualizar código: ${error.message}`);
    }
}

update();
